markdown/3.x/en/documentation/custom-components.md

```diff
@@ -0,0 +1,92 @@
+## Implementing custom components 
+
+---
+FACT-Finder Web Components API can be used to implement custom components in any web framework.
+In a typical scenario, we can use [EventAggregator's](/api/3.x/core-event-aggregator) `addFFEvent()` to query FACT-Finder and [ResultDispatcher's](/api/3.x/core-result-dispatcher) `subscribe()` to receive its response. 
+Below you will find two examples following this approach implementing a product carousel component, one in VueJS and one as a native custom element.
+
+### VueJS
+
+The source code is available [here](https://github.com/FACT-Finder-Web-Components/demos/blob/release/3.x/custom-components/vue-carousel.js).
+Let's take a closer look at the parts where FACT-Finder Web Components API is utilized:
+
+#### 1. `addFFEvent`
+
+We call `addFFEvent()` in the `mounted` hook to trigger FACT-Finder search when the component renders for the first time:
+```javascript
+eventAggregator.addFFEvent({
+    type: 'search',
+    query: carousel.query,
+    topics: function () {
+        return ['carousel-' + carousel._uid];
+    }
+});
+```
+* `type: 'search'` indicates that we want to perform a regular search
+* `query` is populated from the component prop
+* `topics` function defines to which topic the results will be dispatched. In this case, we want this topic to be unique for each component instance. Otherwise, other carousels would interfere with each other
+
+**NOTE**: When designing a production-ready component it is advisable to implement a `query` prop watcher to react to its possible changes with consecutive search requests.
+
+#### 2. `subscribe`
+
+We subscribe to the same topic also in the `mounted` hook:
+```javascript
+carousel.subscriptionKey = resultDispatcher.subscribe('carousel-'  + carousel._uid, function (response) {
+    carousel.records = response.searchResult.records.map(function (record) {
+        return record.record;
+    });
+});
```

Comment:
I feel sufficient time has passed to drop the philosophy of providing IE compatible examples only.
If you are fine with it, I'm okay with ES6+ syntax.

BTW, I think I saw a doubled whitespace.
```suggestion
carousel.subscriptionKey = resultDispatcher.subscribe('carousel-' + carousel._uid, function (response) {
    carousel.records = response.searchResult.records.map(function (record) {
        return record.record;
    });
});
```